xla/service/gpu/runtime3/command_buffer_cmd.h

```diff
@@ -163,6 +164,27 @@ class MemcpyDeviceToDeviceCmd : public CommandBufferCmd {
   int64_t num_bytes_;
 };
 
+//===----------------------------------------------------------------------===//
+// MemallocOnDeviceCmd
+//===----------------------------------------------------------------------===//
+
+class MemallocOnDeviceCmd : public CommandBufferCmd {
+ public:
+  MemallocOnDeviceCmd(int64_t size, BufferAllocation::Slice dst, se::DeviceMemoryBase* dptr);
```

Comment:
I think that only one should be passed as argument:

(1) You either pass `Slice` and inside Record figure out what pointer has to be updated
(2) Or you pass `dptr` and update it.

`Slices slices()` is supposed to return slices that command reads/writes from/to, but for allocation cmd it actually "creates" it.

Reply:
OK, so for allocation command, `slice()` function doesn't need to return the created allocation?

Reply:
I think it should be empty, as I commented below, slice doesn't seem to belong here